xarray/core/accessors.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .common import is_datetime_like
+from .extensions import register_dataarray_accessor
+from .pycompat import dask_array_type
+
+import numpy as np
+import pandas as pd
+
+
+def _get_date_field_from_dask(values, name):
+    """Specialized date field accessor for data contained in dask arrays
+    """
+    from dask import delayed
+    from dask.array import from_delayed, map_blocks
+    from dask.dataframe import from_array
+
+    @delayed
+    def _getattr_from_dask(accessor):
+        attr_values = getattr(accessor, name).values
+        return attr_values.compute()
+
+    def _ravel_and_access(darr):
```

Comment:
This is more complicated than it needs to, as evidenced by the nested calls to compute. The great thing about `map_blocks` is that you can make everything eager, e.g.,
```python
def _ravel_and_access(darr):
    values_as_series = pd.Series(values.ravel())
    field_values = getattr(values_as_series.dt, name).values
    return field_values.reshape(values.shape)
```
That's it!